control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go

```diff
@@ -510,6 +586,31 @@ func (r *Controller) getConsulPartition() string {
 	return r.ConsulPartition
 }
 
+// getWriteCacheKey gets a key to track syncronization of a K8s service to deduplicate writes to Consul.
+// See also WriteCache.hasMatch.
+func getWriteCacheKey(serviceName types.NamespacedName) string {
+	return serviceName.String()
+}
+
+// getWriteHash gets a hash of the given resource to deduplicate writes to Consul.
+//
+// This hash is not intended to be cryptographically secure, only deterministic and collision-resistent
+// for tens of thousands of values.
+//
+// If an error occurs marshalling the resource for the hash, returns a nil hash value and the error.
+// error will be returned.
+func getWriteHash(r *pbresource.Resource) ([]byte, error) {
+	// We convert the entire resource (not just its own Data, which is also an Any)
+	// to Any in order to take advantage of the deterministic marshal offered by proto
+	// and include fields like Meta, which are not part of the resource Data.
+	data := new(anypb.Any)
```

Comment:
There is a toplevel `proto` function you can call to do this without the `anypb` step. We already do that variant in `agent/xds/delta.go` for reference.

Reply:
You beat me to pushing - thanks for the tip! Fixed.